src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3637,21 +3688,43 @@ def create_functionapp(cmd, resource_group_name, name, storage_account, plan=Non
 
     con_string = _validate_and_get_connection_string(cmd.cli_ctx, resource_group_name, storage_account)
 
-    if is_linux:
+    if environment is not None:
+        site_config.app_settings.append(NameValuePair(name='AzureWebJobsStorage', value=con_string))
+        if docker_registry_server_url is not None:
+            site_config.app_settings.append(NameValuePair(name='DOCKER_REGISTRY_SERVER_URL', value=docker_registry_server_url))
+
+        if (not registry_username and not registry_password and
+                docker_registry_server_url and '.azurecr.io' in docker_registry_server_url):
+            logger.warning('No credential was provided to access Azure Container Registry. Trying to look up...')
+            parsed = urlparse(docker_registry_server_url)
+            registry_name = (parsed.netloc if parsed.scheme else parsed.path).split('.')[0]
+            try:
+                registry_username, registry_password = _get_acr_cred(cmd.cli_ctx, registry_name)
+            except Exception as ex:  # pylint: disable=broad-except
+                logger.warning("Retrieving credentials failed with an exception:'%s'", ex)  # consider throw if needed
+
+        if registry_username is not None:
+            site_config.app_settings.append(NameValuePair(name='DOCKER_REGISTRY_SERVER_USERNAME', value=registry_username))
+        if registry_password is not None:
+            site_config.app_settings.append(NameValuePair(name='DOCKER_REGISTRY_SERVER_PASSWORD', value=registry_password))
+
+        app_settings_dict = {}
+        matched_runtime.app_insights = True
+    elif is_linux:
         functionapp_def.kind = 'functionapp,linux'
         functionapp_def.reserved = True
         is_consumption = consumption_plan_location is not None
         if not is_consumption:
             site_config.app_settings.append(NameValuePair(name='MACHINEKEY_DecryptionKey',
                                                           value=str(hexlify(urandom(32)).decode()).upper()))
-            if deployment_container_image_name:
+            if image:
                 functionapp_def.kind = 'functionapp,linux,container'
                 site_config.app_settings.append(NameValuePair(name='DOCKER_CUSTOM_IMAGE_NAME',
-                                                              value=deployment_container_image_name))
+                                                              value=image))
```

Comment:
Yeah so this is under `elif is_linux` so if the customer is creating a Centauri-related function app, this logic will not be run

Reply:
I see we set it to `true` at https://github.com/Azure/azure-cli/blob/a350ac2b82f8d9985ef9b85e2655e834985aeadd/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L3664, so even for centauri app the logics will be applied right?

Reply:
No - `if environment is not None` is evaluated first and if it is true, only Centauri related logic will run and the code under `elif is_linux` would not be run.

Reply:
didn't see it's a `else if`. thanks.

Reply:
We're really only setting is_linux =True because of 
```
runtime_helper = _FunctionAppStackRuntimeHelper(cmd, linux=is_linux, windows=(not is_linux))
matched_runtime = runtime_helper.resolve("dotnet" if not runtime else runtime,
                                             runtime_version, functions_version, is_linux)
```